Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -644,6 +626,67 @@ public final class SwiftTargetBuildDescription {
 
         return args
     }
+    
+    /// Determines the arguments needed to run `swift-symbolgraph-extract` for
+    /// this module.
+    public func symbolGraphExtractArguments() throws -> [String] {
+        var args = [String]()
+        args += try self.cxxInteroperabilityModeArguments(
+            propagateFromCurrentModuleOtherSwiftFlags: true)
+        return args
+    }
+
+    // FIXME: this function should operation on a strongly typed buildSetting
+    // Move logic from PackageBuilder here.
+    /// Determines the arguments needed for cxx interop for this module.
+    func cxxInteroperabilityModeArguments(
+        // FIXME: Remove argument
+        // This argument is added as a stop gap to support generating arguments
+        // for tools which currently don't leverage "OTHER_SWIFT_FLAGS". In the
+        // fullness of time this function should operate on a strongly typed
+        // "interopMode" property of SwiftTargetBuildDescription instead of
+        // digging through "OTHER_SWIFT_FLAGS" manually.
+        propagateFromCurrentModuleOtherSwiftFlags: Bool
+    ) throws -> [String] {
+        func cxxInteroperabilityModeAndStandard(
+            for module: ResolvedModule
+        ) -> [String]? {
+            let scope = self.buildParameters.createScope(for: module)
```

Comment:
Is this correct to use `buildParameters` of the context here for dependency scopes? What if if this i.e. executable target that has a plugin or a macro dependency and that in turn has other dependencies, we'd use "destination" parameters for their scope, which seems counter-intuitive? 

It seems like we need to avoid adding new uses of `recursiveTargetDependencies()` in this context when build parameters are involved, because they are strongly connected to a destination, and instead traverse a build plan graph and use `TargetDependencyDescription`.

Reply:
I'd rather not change this logic here, im simply moving the existing code such that symbol graph can use it.

Reply:
I guess the point of that comment is to say, we should not throw away the typed interop information at that layer and try to recover it here by parsing flags, we should instead determine the flags at this layer

Reply:
Maybe you could do the same thing as `.SWIFT_VERSION` and add a new declaration for assigments like that which means that you won't need to deal with prefixes here?

Reply:
I don't mean it has to be in this PR, just an idea how to avoid having to prefix match things...